### ISSUE TITLE
[BUG] 502 error for leantime.my.domain/dashboard/home when running the docker container

### ISSUE THREAD

**Issue by Aephir** [Leantime/leantime#1792]
*If you have problems with the installation please use our [community forum](https://community.leantime.io)*

**What set up are you using**
* [ ] Cloud Hosted Version
* [x] Self Hosted (Docker)

**Describe the bug**
I get to the login page if I navigate to `leantime.my.domain`, but after logging in, I get forwarded to leantime.my.domain/dashboard/home which shows the error `502 Bad Gateway`. If I manually re-enter `leantime.my.domain`, I get the the dashbaord and can use leantime, but since I'm trying to build an app with [nativefier](https://github.com/nativefier/nativefier), this won't work anymore.

I can't remember when this behavior started (a while ago, but it has worked fine for months before that) since I just manually navigated back to `leantime.my.domain` instead of trying to fix.

I tried setting `VIRTUAL_PORT: 80` as per [this comment](https://github.com/Leantime/leantime/issues/273#issuecomment-636493319).

The solution in #1341 is less useful in docker (since any pull of image would "undo").

**To Reproduce**
1. Set up leantime and db i docker-compose file (see config below)
2. Set up nginx as reverse proxy (I am assuming this isn't the issue, as this part used to work fine).
3. `docker-compose up -d`
4. navigate to fqdn and log in.
5. See error 502 when directed to fqdn/dashboard/home

<details>

<summary> leantime and db config in docker-compose.yaml </summary>

```
  leantime_db:
    image: mysql:8.0
    container_name: leantime_mysql
    volumes:
      - /mnt/data/docker/leantime/leantime_db:/var/lib/mysql
    restart: unless-stopped
    environment:
      MYSQL_ROOT_PASSWORD: 'RECACTED'
      MYSQL_DATABASE: 'leantime'
      MYSQL_USER: 'admin'
      MYSQL_PASSWORD: 'RECACTED'
    command: --character-set-server=utf8 --collation-server=utf8_unicode_ci

  leantime:
    image: leantime/leantime:latest
    container_name: leantime
    restart: unless-stopped
    environment:
      # LEAN_APP_URL: 'https://leantime.my.domain'                     # Only needed for subdirectory setup; protocol (http or https) and base URL , trailing slash not needed
      LEAN_SITENAME: 'Leantime'                                         # Name of your site, can be changed later
      LEAN_DB_HOST: 'leantime_mysql'                                    # Database host, derived from container_name in leantime_db container
      LEAN_DB_USER: 'admin'
      LEAN_DB_PASSWORD: 'RECACTED'
      LEAN_DB_DATABASE: 'leantime'
      LEAN_DEFAULT_TIMEZONE: 'Europe/Stockholm'                         # Set default server timezone
      LEAN_SESSION_PASSWORD: 'RECACTED'         # Salting sessions. Replace with a strong password
      LEAN_SESSION_EXPIRATION: 28800                                    # How many seconds after inactivity should we logout?  28800seconds = 8hours
    volumes:
      - /mnt/data/docker/leantime/public_userfiles:/var/www/html/public/userfiles
      - /mnt/data/docker/leantime/userfiles:/var/www/html/userfiles
    ports:
      - "10.0.30.21:8004:80"                                                       # The port to expose and access Leantime
    depends_on:
      - leantime_db                                                     # Don't start Leantime unless leantime_db is running
```

</details>

<details>

<summary> leantime-specific nginix conf (rest is standard and unchanged from linuxerver swag container) </summary>

```
server {
        listen 443 ssl;
        # listen 80;

        root /config/www;
        index index.html index.htm index.php;

        server_name leantime.my.domain

        include /config/nginx/ssl.conf;

        client_max_body_size 0;

        location / {
        #               auth_basic "Restricted";
        #               auth_basic_user_file /config/nginx/.htpasswd;
                        include /config/nginx/proxy.conf;
                proxy_pass http://10.0.30.21:8004;
        }
}
```
</details>

I haven't changed anything in either `docker-compose.yaml` or nginx's `leantime.conf` since initial setup, the only thing I have done is pull new images every now and then.

**Expected behavior**
Not getting a 502 error (like things were a few months ago).

**Leantime Version**
Not sure where to see (can't `docker exec -it leantime sh` and use `leantime --version`), but I did a pull of the leantime/leantime:latest today (August 3rd, 2023).

**Server**
nginx

**PHP / MySQL Version**
PHP v8.1.21 (what's in the docker image)
Mysql v8.0.34

**Comment by swissinside** [comment]
I have exactly the same Problem (without Docker) and with the latest release (v2.3.27).

**Comment by thereiam** [comment]
Thank you for the workaround!

I had the same issue but running leantime docker on k8s, below is the nginx-ingress configuration file:

```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: leantime-ingress
  annotations:
    nginx.ingress.kubernetes.io/proxy-buffering: "on"
    nginx.ingress.kubernetes.io/proxy-buffers-number: "4"
    nginx.ingress.kubernetes.io/proxy-buffer-size: "32k"
spec:
  ingressClassName: nginx
  tls:
  - hosts:
      - leantime.mydomain.com
    secretName: leantime-tls
  rules:
  - host: leantime.mydomain.com
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: leantime-service
            port:
              number: 8080
```

**Comment by EHRETic** [comment]
Hi there,

Have the same issue with 2.3.27 on Docker and NGINX as reverse  (didn't had that with previous version)

I'll try the buffer fix. 😉👌

**Comment by marcelfolaron** [comment]
Thank you for the report and fix submitted. I'll take a look to see why the header is so big and/or add the buffer update to our nginx onfig.

**Comment by Emad-Marji** [comment]
> I had exactly the same problem (without Docker) and with the latest release (v2.3.27). In my case the error in the logs was `upstream sent too big header while reading response header from upstream`. After I added the following additional nginx directives it worked:
> 
> ```
> proxy_buffers 8 16k;
> proxy_buffer_size 32k;
> ```

How to add those and where?

**Comment by Aephir** [comment]
> How to add those and where?

@Emad-Marji, that depends slightly on your nginx is set up, but most likely in the file you use in the `sites-conf` directory to configure leantime (a guess would be that it's called `leantime.conf`).

I have my ssl and proxy "default" configuration in other files (and use include), so my `config/nginx/site-confs/leantime.conf` is pretty minimal, but I add it to that file like this:

```
server {
        listen 443 ssl;

        root /config/www;
        index index.html index.htm index.php;

        server_name leantime.MY.DOMAIN;

        include /config/nginx/ssl.conf;

        client_max_body_size 0;
        proxy_buffers 8 16k;
        proxy_buffer_size 32k;

        location / {
                include /config/nginx/proxy.conf;
                proxy_pass http://10.0.30.21:8004;
        }
}
```

of course `MY.DOMAIN` points to my actual domain :)

**Comment by swissinside** [comment]
> How to add those and where?

I have Plesk installed on my server. If this is the case for you @Emad-Marji, then you can simply add these two directives in the `Apache & nginx Settings` in the `Additional nginx directives` section.

**Comment by marcelfolaron** [comment]
2.4-beta was just released and might fix this issue as well as we reduced header size across the board and improved performance

**Comment by Emad-Marji** [comment]
Thanks @swissinside,
Actually, I installed it as a docker container in my Synology NAS and it is working perfectly fine as long as I am accessing it locally via an IP:PORT. When configuring the Synology Reverse Proxy with Lets Encrypt to access it from outside, i managed to do that but without the Dashboard. Sometimes it gives 502 error and others a Synology Webstation File Not Found. All other pages are accessible except the dashboard. 
I reinstalled it again as a web app using PHP 8.2, and MariaDB 10 but unfortunately, the same results (accessible locally, but without a dashboard externally).
I tried @Aephir solution and managed to add the buffer directives after consultation with Synology as I believe their Reverse Proxy is based on nginx. The same results, unfortunately.
I have just updated the docker container to use the 2.4-beta with the same configuration. It is working internally but seems to miss the CSS styles when accessed from outside. I may need to redeploy it in a clean environment instead of updating the docker-compose to use the beta one.
So far, it seems to be a Synology issue as I have no issue deploying it as a container under Debian 12 and Nginx Reverse Proxy.

**Comment by swissinside** [comment]
> 2.4-beta was just released and might fix this issue as well as we reduced header size across the board and improved performance

In my case it is even worse with 2.4-beta. I had to go back to v2.3.27. Does anyone else has problems with 2.4-beta?

**Comment by marcelfolaron** [comment]
Header size is bigger?

**Comment by Emad-Marji** [comment]
2.4-beta works perfectly for me. All issues are resolved. with and without docker

**Comment by swissinside** [comment]
> Header size is bigger?

@marcelfolaron: Sorry, I misread the logs. The problem is no longer related to the header size but to the caching of a view. As @Emad-Marji stated this issue seems to be resolved.

**Comment by Emad-Marji** [comment]
Apologies dears, but I am facing a new issue with deploying 2.4-beta in Docker. This is related to "Blocked loading mixed active content" over HTTPS, which blocks CSS and JS files. I am not facing this with the previous version 2.3.27

Am I doing something wrong? Appreciate your advice.

Here is the docker-compose for easy reference. I have also attached 2 images from localhost (with no issue) and a test sub-domain (which is calling css/js/... from http instead of https). 

```
version: "3.9"
services:
    leandb:
        image: mariadb:latest
        container_name: mysql_lean
        volumes:
            - ./db_data:/var/lib/mysql
        environment:
            MYSQL_ROOT_PASSWORD: 321.qwerty
            MYSQL_USER: admin
            MYSQL_PASSWORD: 321.qwerty
            MYSQL_DATABASE: lean_db
        restart: unless-stopped
        ports:
            - 3336:3306
        networks:
            - lean-net
        command: --character-set-server=utf8 --collation-server=utf8_unicode_ci
    leanapp:
        image: leantime/leantime:latest
        #image: leantime/leantime:2.4-beta
        container_name: leantime_app
        volumes:
            - ./public_userfiles:/var/www/html/public/userfiles
            - ./userfiles:/var/www/html/userfiles
        ports:
            - 33000:80
        networks:
            - lean-net
        environment:
            LEAN_DB_HOST: mysql_lean
            LEAN_DB_DATABASE: lean_db
            LEAN_DB_USER: admin
            LEAN_DB_PASSWORD: 321.qwerty
        depends_on:
            - leandb  
volumes:
    db_data:
    public_userfiles:
    userfiles:
networks:
  lean-net:
```

![Snag_134995d9](https://github.com/Leantime/leantime/assets/102861171/0f452404-2c62-405a-afa6-9f03b2ff7174)
![Snag_1349c12f](https://github.com/Leantime/leantime/assets/102861171/76f515f1-6e51-4a02-a608-ea33fe38b51b)

**Comment by marcelfolaron** [comment]
Thanks for the detailed report. Did you set the LEAN_APP_URL (including https)

Question to all: How do other (PHP) applications handle the reverse proxy set up? I am somewhat at a loss because in my limited knowledge of reverse proxies, it should work. Laravel and Symfony require you to set a application URL. They allow you to set trusted proxies which I'd be happy to set up but that shouldn't cause the errors you're all seeing.

**Comment by EHRETic** [comment]
> Thanks for the detailed report. Can you set the LEAN_APP_URL (your domain including https) in your composer config?
> 
> Question to all: How do other (PHP) applications handle the reverse proxy set up? I am somewhat at a loss because in my limited knowledge of reverse proxies, it should work. Laravel and Symfony require you to set a application URL. They allow you to set trusted proxies which I'd be happy to set up but that shouldn't cause the errors you're all seeing.

I have to admit that I'd like to have this "fixed" too because in order to have an instance available with an internal and an external URL/FQDN, I had to create 2 docker instances pointing to the same DB.
Not something very bothering because it works fine and the flexibility of Docker helps a lot.

To the ones specialized in the field, could that be related because Leantime use "non relative links" (URLs built entirely from the configured DNS name instead of building them relatively from the home of the same DNS?

I can do (a lot of) tests if required (create test instances, etc..) 😉

**Comment by Emad-Marji** [comment]
You're a rockstar @marcelfolaron 
After adding my domain to LEAN_APP_URL, it is working over https now.... and yes, not working in http, but who cares; sorry @EHRETic, I am joking  😉

I did not pay attention to this as it is mentioned as optional in the sample.env.... also, in version 2.3.27, both HTTP and HTTPS are working without specifying an APP_URL value.

`LEAN_APP_URL = ''                                  # Base URL, only needed for subfolder installation
`

In addition to the docker deployment, I have used the Synology Web Station and created 2 profiles for Apache and Nginx. I have installed 2 instances of Leantime each using a different profile and both are working as expected.

![Snag_13826e28](https://github.com/Leantime/leantime/assets/102861171/0a42b2bb-0e4d-4a26-91c8-c61dedfd5302)
![Snag_1382852b](https://github.com/Leantime/leantime/assets/102861171/267847b6-5299-4b82-9867-21928a8233d8)

**Comment by Emad-Marji** [comment]
One more question if you allow me:
Using Docker, the maximum allowed file size is 2MB. Is there a way to change that? Note that I can ONLY access the mounted volumes (userfiles, public_userfiles and DB) in Synology.
With normal installation (not docker), I successfully tested an upload of an 18MB file with no issue.

**Comment by marcelfolaron** [comment]
@Emad-Marji Glad it is working now. Regarding the file size limit: I will have to add that to the config of the docker image. In the meantime you could log into your container and change it in there through the php.ini file. 

@EHRETic I'll do some more digging. Without the AppUrl this will require some additional work as I have to capture the request from the proxy.

**Comment by Emad-Marji** [comment]
Appreciated again dear @marcelfolaron,
I found your answer re post# #952 and I have managed to increase the size to 20MB
`/usr/local/etc/php/conf.d/custom.ini`

**Comment by EHRETic** [comment]
Hi there,

Just to mention that 2.4-beta solved the header issue on my test instance, nice 😉👌
